### ISSUE TITLE
from_pandas_dataframe bug with MultiDiGraph and edge_attr

### ISSUE THREAD

**Issue by jabellcu** [networkx/networkx#2374]
from_pandas_dataframe fails when building a MultiDiGraph reading edge attributes. Please see code below:

```
import pandas as pd
df = pd.DataFrame.from_items([
        ( 'O', ['X1', 'X1', 'X1', 'X1', 'Y1', 'Y1', 'Y1', 'Y1', 'Z1', 'Z1', 'Z1', 'Z1']),
        ( 'D', ['X4', 'X4', 'X4', 'X4', 'Y3', 'Y3', 'Y3', 'Y3', 'Z3', 'Z3', 'Z3', 'Z3']),
        ('St', ['X1', 'X2', 'X3', 'X4', 'Y1', 'Y2', 'X2', 'Y3', 'Z1', 'X3', 'Z2', 'Z3']),
        ('Co', ['zA', 'zB', 'zB', 'zB', 'zC', 'zC', 'zC', 'zC', 'zD', 'zD', 'zE', 'zE']),
        ('Mi', [   0,   54,   49,   44,    0,   34,   29,   24,    0,   14,    9,   4])])

import networkx as nx


### Works perfectly
G = nx.from_pandas_dataframe(df, source='O', target='D',
                             create_using=nx.MultiDiGraph())


### Doesn't work
G = nx.from_pandas_dataframe(df, source='O', target='D',
                             edge_attr=['St','Co','Mi'],
                             create_using=nx.MultiDiGraph())


---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-1cfba94ebbd4> in <module>()
      1 #Doesn't work
      2 G = nx.from_pandas_dataframe(df, source='O', target='D',
----> 3                              edge_attr=['St','Co','Mi'], create_using=nx.MultiDiGraph())

//anaconda/lib/python2.7/site-packages/networkx/convert_matrix.py in from_pandas_dataframe(df, source, target, edge_attr, create_using)
    209         # Iteration on values returns the rows as Numpy arrays
    210         for row in df.values:
--> 211             g.add_edge(row[src_i], row[tar_i], {i:row[j] for i, j in edge_i})
    212 
    213     # If no column names are given, then just return the edges.

//anaconda/lib/python2.7/site-packages/networkx/classes/multidigraph.py in add_edge(self, u, v, key, attr_dict, **attr)
    342             datadict.update(attr_dict)
    343             keydict = self.edge_key_dict_factory()
--> 344             keydict[key] = datadict
    345             self.succ[u][v] = keydict
    346             self.pred[v][u] = keydict

TypeError: unhashable type: 'dict'
```

**Comment by hagberg** [comment]
Thanks for the report.  This works correctly for me with the master branch of networkx in the github repository.   Could you try that version?

```python
In [7]: G = nx.from_pandas_dataframe(df, source='O', target='D',
   ...:                              edge_attr=['St','Co','Mi'],
   ...:                              create_using=nx.MultiDiGraph())

In [8]: list(G.edges(data=True))
Out[8]: 
[('X1', 'X4', {'Co': 'zA', 'Mi': 0, 'St': 'X1'}),
 ('X1', 'X4', {'Co': 'zB', 'Mi': 54, 'St': 'X2'}),
 ('X1', 'X4', {'Co': 'zB', 'Mi': 49, 'St': 'X3'}),
 ('X1', 'X4', {'Co': 'zB', 'Mi': 44, 'St': 'X4'}),
 ('Y1', 'Y3', {'Co': 'zC', 'Mi': 0, 'St': 'Y1'}),
 ('Y1', 'Y3', {'Co': 'zC', 'Mi': 34, 'St': 'Y2'}),
 ('Y1', 'Y3', {'Co': 'zC', 'Mi': 29, 'St': 'X2'}),
 ('Y1', 'Y3', {'Co': 'zC', 'Mi': 24, 'St': 'Y3'}),
 ('Z1', 'Z3', {'Co': 'zD', 'Mi': 0, 'St': 'Z1'}),
 ('Z1', 'Z3', {'Co': 'zD', 'Mi': 14, 'St': 'X3'}),
 ('Z1', 'Z3', {'Co': 'zE', 'Mi': 9, 'St': 'Z2'}),
 ('Z1', 'Z3', {'Co': 'zE', 'Mi': 4, 'St': 'Z3'})]
```

**Comment by jabellcu** [comment]
**You are right!** master branch works seamlessly.

I have been able to reproduce the bug in networkx 1.11, which seem to be the latest available version in conda for python 3.5.

This might hopefully help others.

Many thanks.
